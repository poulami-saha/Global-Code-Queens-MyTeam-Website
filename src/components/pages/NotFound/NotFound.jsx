import React from "react";
import Typography from "../../utility/Typography/Typography";
import styles from "./NotFound.module.css";

const NotFound = () => {
  return (
    <section className={styles.NotFound_container}>
    <article className={styles.NotFound_content}>
      <Typography elType="h1Small" className={styles.NotFound_heading}>Page not found</Typography>
      <Typography elType="body1">We can’t seem to find the page you’re looking for.<br/>
      Try going back to the previous page.</Typography>
      </article>
      </section>
  );
};

export default NotFound;
