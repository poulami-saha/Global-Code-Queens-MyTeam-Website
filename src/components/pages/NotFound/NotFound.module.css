/* NotFound container */
.NotFound_container {
  display: grid;
  place-items: center;
  height: 100vh;
  background-color: var(--deep-jungle-blue);
  color: var(--primary-white);
  position: relative;
  z-index: -2;
}

/* NotFound container  background decoration*/
.NotFound_container::before,
.NotFound_container::after {
  content: "";
  position: absolute;
  z-index: -1;
}

.NotFound_container::before {
  width: 200px;
  height: 200px;
  top: 62px;
  left: -100px;
  background-image: url("../../../assets/bg-pattern-about-2-contact-1.svg");
}

.NotFound_container::after {
  width: 200px;
  height: 112px;
  bottom: 0px;
  right: -100px;
  background-image: url("../../../assets/bg-pattern-contact-2.svg");
}

@media (min-width: 768px) {
  .NotFound_container::before {
    top: 124px;
  }
  .NotFound_container::after {
    bottom: 0px;
  }
}

/* NotFound content */
.NotFound_content {
  padding: 24px;
  text-align: center;
}
.NotFound_heading {
  margin-bottom: 40px;
}
