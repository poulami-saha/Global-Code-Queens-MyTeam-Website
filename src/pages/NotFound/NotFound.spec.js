import React from "react";
import { render } from "@testing-library/react";
import NotFound from "./NotFound";

test("renders NotFound correctly and matches snapshot", async () => {
  const { asFragment, getByText } = render(<NotFound />);

  // Snapshot test
  expect(asFragment()).toMatchSnapshot();

  // Content test
  expect(getByText("Page not found")).toBeInTheDocument();
  expect(
    getByText(/We can’t seem to find the page you’re looking for./)
  ).toBeInTheDocument();
  expect(getByText(/Try going back to the previous page./)).toBeInTheDocument();
});
