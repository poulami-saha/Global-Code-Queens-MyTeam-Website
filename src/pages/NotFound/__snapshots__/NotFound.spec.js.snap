// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NotFound correctly and matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="NotFound_container"
  >
    <article
      class="NotFound_content"
    >
      <h1
        class="NotFound_heading h1Small"
        id=""
      >
        Page not found
      </h1>
      <p
        class="body1"
        id=""
      >
        We can’t seem to find the page you’re looking for.
        <br />
        Try going back to the previous page.
      </p>
    </article>
  </section>
</DocumentFragment>
`;
